manager/controllers/baseline_update.go

```diff
@@ -0,0 +1,152 @@
+package controllers
+
+import (
+	"app/base"
+	"app/base/database"
+	"app/base/models"
+	"app/base/utils"
+	"app/manager/middlewares"
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	"github.com/pkg/errors"
+)
+
+type UpdateBaselineRequst struct {
+	Name         string            `json:"name"`
+	InventoryIDs map[string]string `json:"inventory_ids"`
+	Config       BaselineConfig    `json:"config"`
+}
+
+type BaselineConfig struct {
+	ToTime string `json:"to_time"`
+}
+
+type AccountID struct {
+	RhAccountID int
+}
+type UpdateBaselineResponse struct {
+	BaselineID int
+}
+
+// @Summary Update a baseline for my set of systems
+// @Description Update a baseline for my set of systems
+// @ID updateBaseline
+// @Security RhIdentity
+// @Accept   json
+// @Produce  json
+// @Param    baseline_id path    int                  true "Baseline ID"
+// @Param    body        body    UpdateBaselineRequst true "Request body"
+// @Success 200 {object} UpdateBaselineResponse
+// @Router /api/patch/v1/baselines/{baseline_id} [post]
+func BaselineUpdateHandler(c *gin.Context) {
+	account := AccountID{
+		RhAccountID: c.GetInt(middlewares.KeyAccount),
+	}
+
+	var req UpdateBaselineRequst
+	if err := c.ShouldBindJSON(&req); err != nil {
+		LogAndRespBadRequest(c, err, "Invalid request body")
+	}
+
+	baselineID := c.Param("baseline_id")
+	if baselineID == "" {
+		c.JSON(http.StatusBadRequest, utils.ErrorResponse{Error: "baseline_id param not found"})
+		return
+	}
+
+	var exists int64
+	err := database.Db.Model(&models.Baseline{}).
+		Where("id = ? ", baselineID).Count(&exists).Error
+	if err != nil {
+		LogAndRespError(c, err, "database error")
+	}
+	if exists == 0 {
+		LogAndRespNotFound(c, errors.New("Baseline not found"), "Baseline not found")
+		return
+	}
+
+	config, err := json.Marshal(&req.Config)
+	if err != nil {
+		LogAndRespError(c, err, "Invalid config")
+	}
+
+	ID, err := strconv.Atoi(baselineID)
+	if err != nil {
+		LogAndRespError(c, err, "Invalid baseline id")
+	}
+
+	baseline := models.Baseline{
+		ID:          ID,
+		Name:        req.Name,
+		Config:      config,
+		RhAccountID: account.RhAccountID,
+	}
+
+	err = buildUpdateBaselineQuery(baseline, req.InventoryIDs, account, c)
```

Comment:
We can simplify that.
1) You don't need to create baseline object. Just pass req to the buildBaselineQuery. You can convert config to string within the function.
2) Don't pass "c" to the function. Just return error and use "LogAndError" just once on the line 89.